Description: enabled OpenGL on the dekstop instead of OpenGL|ES
Author: Stephen M. webb <stephen.webb@canonical.om>
Bug-Ubuntu: https://bugs.launchpad.net/qtubuntu/+bug/1274763

--- a/src/platforms/base/context.cc
+++ b/src/platforms/base/context.cc
@@ -37,6 +37,15 @@
 }
 #endif
 
+static EGLenum api_in_use()
+{
+#ifdef QTUBUNTU_USE_OPENGL
+  return EGL_OPENGL_API;
+#else
+  return EGL_OPENGL_ES_API;
+#endif
+}
+
 QUbuntuBaseContext::QUbuntuBaseContext(QUbuntuBaseScreen* screen) {
   DASSERT(screen != NULL);
   eglDisplay_ = screen->eglDisplay();
@@ -47,7 +56,7 @@
   attribs.append(EGL_CONTEXT_CLIENT_VERSION);
   attribs.append(2);
   attribs.append(EGL_NONE);
-  ASSERT(eglBindAPI(EGL_OPENGL_ES_API) == EGL_TRUE);
+  ASSERT(eglBindAPI(api_in_use()) == EGL_TRUE);
   ASSERT((eglContext_ = eglCreateContext(
       eglDisplay_, screen->eglConfig(), EGL_NO_CONTEXT, attribs.constData())) != EGL_NO_CONTEXT);
 
@@ -64,10 +73,10 @@
   DASSERT(surface->surface()->surfaceType() == QSurface::OpenGLSurface);
   EGLSurface eglSurface = static_cast<QUbuntuBaseWindow*>(surface)->eglSurface();
 #if defined(QT_NO_DEBUG)
-  eglBindAPI(EGL_OPENGL_ES_API);
+  eglBindAPI(api_in_use());
   eglMakeCurrent(eglDisplay_, eglSurface, eglSurface, eglContext_);
 #else
-  ASSERT(eglBindAPI(EGL_OPENGL_ES_API) == EGL_TRUE);
+  ASSERT(eglBindAPI(api_in_use()) == EGL_TRUE);
   ASSERT(eglMakeCurrent(eglDisplay_, eglSurface, eglSurface, eglContext_) == EGL_TRUE);
   printOpenGLESConfig();
 #endif
@@ -77,10 +86,10 @@
 void QUbuntuBaseContext::doneCurrent() {
   DLOG("QUbuntuBaseContext::doneCurrent (this=%p)", this);
 #if defined(QT_NO_DEBUG)
-  eglBindAPI(EGL_OPENGL_ES_API);
+  eglBindAPI(api_in_use());
   eglMakeCurrent(eglDisplay_, EGL_NO_SURFACE, EGL_NO_SURFACE, EGL_NO_CONTEXT);
 #else
-  ASSERT(eglBindAPI(EGL_OPENGL_ES_API) == EGL_TRUE);
+  ASSERT(eglBindAPI(api_in_use()) == EGL_TRUE);
   ASSERT(eglMakeCurrent(eglDisplay_, EGL_NO_SURFACE, EGL_NO_SURFACE, EGL_NO_CONTEXT) == EGL_TRUE);
 #endif
 }
@@ -89,10 +98,10 @@
   // DLOG("QUbuntuBaseContext::swapBuffers (this=%p, surface=%p)", this, surface);
   EGLSurface eglSurface = static_cast<QUbuntuBaseWindow*>(surface)->eglSurface();
 #if defined(QT_NO_DEBUG)
-  eglBindAPI(EGL_OPENGL_ES_API);
+  eglBindAPI(api_in_use());
   eglSwapBuffers(eglDisplay_, eglSurface);
 #else
-  ASSERT(eglBindAPI(EGL_OPENGL_ES_API) == EGL_TRUE);
+  ASSERT(eglBindAPI(api_in_use()) == EGL_TRUE);
   ASSERT(eglSwapBuffers(eglDisplay_, eglSurface) == EGL_TRUE);
 #endif
 }
@@ -100,9 +109,9 @@
 void (*QUbuntuBaseContext::getProcAddress(const QByteArray& procName)) () {
   DLOG("QUbuntuBaseContext::getProcAddress (this=%p, procName=%s)", this, procName.constData());
 #if defined(QT_NO_DEBUG)
-  eglBindAPI(EGL_OPENGL_ES_API);
+  eglBindAPI(api_in_use());
 #else
-  ASSERT(eglBindAPI(EGL_OPENGL_ES_API) == EGL_TRUE);
+  ASSERT(eglBindAPI(api_in_use()) == EGL_TRUE);
 #endif
   return eglGetProcAddress(procName.constData());
 }
